common/base/include/claragenomics/utils/device_preallocated_allocator.cuh

```diff
@@ -78,14 +82,16 @@ public:
     /// Memory allocation is aligned by 256 bytes
     /// \param ptr on return pointer to allocated memory, nullptr if allocation was not successful
     /// \param bytes_needed
-    /// \param associated_stream on deallocation this block will be free only once all work in this stream has finished
+    /// \param associated_streams on deallocation this block will be freed only once all previosly scheduled work in these streams has finished, using default stream as default
     /// \return cudaSuccess if allocation was successful, cudaErrorMemoryAllocation otherwise
     cudaError_t DeviceAllocate(void** ptr,
                                size_t bytes_needed,
-                               cudaStream_t associated_stream = 0)
+                               const std::vector<cudaStream_t>& associated_streams = {{0}})
```

Comment:
Do we really need to provide the default argument for any interface compatibility here? Otherwise I suggest to drop it and let the caller handle this explicitly.

Reply:
@mimaric

Reply:
Hmm, yes, here it makes sense to drop the default argument. I thought it's the same as the comment above, but unlike `CachingDeviceAllocator` `DevicePreallocatedAllocator` does not get called by anyone else.

Reply:
done